### PR TITLE
Basic spanish support

### PR DESCRIPTION
**Pull request by Larision**
Basic support to Spain country, Open_positions, full_tax_report and a copied us_tax_report. I will try do an expecific spanish_tax_report but hard to me code it. ill try do my best...

### CODE REVIEW

File: src/rp2/plugin/country/es.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    def get_long_term_capital_gain_period(self) -> int:
+        return 365
+
+    # Default accounting method to use if the user doesn't specify one on the command line
+    def get_default_accounting_method(self) -> str:
+        return "fifo"
+
+    # Set of accounting methods accepted in the country
+    def get_accounting_methods(self) -> Set[str]:
+        # Temporarily removed lifo and hifo due to https://github.com/eprbell/rp2/issues/79
```
Comment (eprbell) [inline src/rp2/plugin/country/es.py:37]: Curious: does Spanish tax law accept lifo and hifo?

**Review by eprbell — CHANGES_REQUESTED** [review]
Nice! I think this has value even just with open_positions and rp2_full_report. Here are a few general comments:
* es.tax_report_es is supposed to be an accountant friendly report: something that Spanish tax professional would immediately be able to recognize and use (currently it mimics the US form 8949, which doesn't reflect Spanish taxes). So I think you have two options:
  * rewrite it as described above, or
  * remove it altogether for now and leave it for future development (you can remove it for es.py, inside function `get_report_generators`).
* update the documentation (look for "japan" and add a Spain-specific snippet)

BTW, do you know what is the name of the accountant friendly Spanish tax form (equivalent of US 8949)?

Thanks for contributing to RP2!

File: src/rp2/plugin/country/es.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 eprbell
```
[the remark this answers is not in the document]
Comment (Larision) [reply src/rp2/plugin/country/es.py:1]: Srry, i´m really a big newbie in all about code standards, python, git..., i´ll try learn some about basis. Fixing them.
Thanks for tips.

File: src/rp2/plugin/country/es.py (hunk 1 above)
Comment (Larision) [reply src/rp2/plugin/country/es.py:37]: No, just FIFO

**Comment by Larision** [comment]
> es.tax_report_es is supposed to be an accountant friendly report: something that Spanish tax professional would immediately be able to recognize and use (currently it mimics the US form 8949, which doesn't reflect Spanish taxes). So I think you have two options:
> rewrite it as described above, or
> remove it altogether for now and leave it for future development (you can remove it from es.py, function get_report_generators).

Removed tax_report_es
Spain dont have any accountant friendly tax form so better i remove report.

> update the documentation (look for "japan" or "jp" and add a Spain-specific snippet)

Checking and adding it

Thank you

File: docs/output_files.md [hunk 3]
```diff
@@ -76,7 +77,7 @@ The NTA (National Tax Agency) provides an excel spreadsheet for calculating the
 The summary sheet contains the total for the tax year. This is what is normally reported as "miscellaneous income" (雑所得). However, it may be reported as business income in certain circumstances. Please do your own research.
 
 ## Unrealized Gains (Open Positions Report Output)
-The open_positions_us.ods output file is an ODS-format spreadsheet designed to provide information on assets with non-zero crypto balances including gains and losses.
+The open_positions.ods output file is an ODS-format spreadsheet designed to provide information on assets with non-zero crypto balances including gains and losses.
```
Comment (eprbell) [inline docs/output_files.md:80]:
```suggestion
The open_positions_\*.ods output file is an ODS-format spreadsheet designed to provide information on assets with non-zero crypto balances including gains and losses.
```

File: docs/user_faq.md [hunk 4]
```diff
@@ -85,13 +85,13 @@ In some cases exchange reports miss spot price information. In such situations y
 The user adds the tokens to the `assets` field of the [config file](input_files.md#the-config-file): RP2 accepts as valid all the tokens present in this field. See also the question on [writing a config file from scratch](#can-i-avoid-writing-a-config-file-from-scratch).
 
 ### What Accounting Methods Are Supported?
-Currently only the [FIFO](https://www.investopedia.com/terms/f/fifo.asp) accounting method is supported for US taxes. Only the Total Average Method is supported for Japanese taxes. 
+Currently only the [FIFO](https://www.investopedia.com/terms/f/fifo.asp) accounting method is supported for US and ES taxes. Only the Total Average Method is supported for Japanese taxes.
```
Comment (eprbell) [inline docs/user_faq.md:88]:
```suggestion
Currently only the [FIFO](https://www.investopedia.com/terms/f/fifo.asp) accounting method is supported for US and Spanish taxes. Only the Total Average Method is supported for Japanese taxes. 
```

File: src/rp2/plugin/country/es.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        return "fifo"
+
+    # Set of accounting methods accepted in the country
+    def get_accounting_methods(self) -> Set[str]:
+        return {"fifo"}
+
+    # Default set of generators to use if the user doesn't specify them on the command line
+    def get_report_generators(self) -> Set[str]:
+        return {
+            "es.open_positions",
```
Comment (eprbell) [inline src/rp2/plugin/country/es.py:42]: I don't think you don't need a Spain-specific version of open_positions (see the [Japanese plugin](https://github.com/eprbell/rp2/blob/main/src/rp2/plugin/country/jp.py#L45) as an example):
```suggestion
            "open_positions",
```

This means you can also delete `src/rp2/plugin/report/es/open_positions.py` from this PR.

File: src/rp2/plugin/report/es/open_positions.py [hunk 6]
```diff
@@ -0,0 +1,544 @@
+# Copyright 2023 Larision
```
Comment (eprbell) [inline src/rp2/plugin/report/es/open_positions.py:1]: I think this file can be deleted (see my other comment above): you should be able to use the common open_positions.py file.

**Review by eprbell — CHANGES_REQUESTED** [review]
This is coming together nicely: it's great to see support for the third country in RP2 (Spain, after Japan and US) being added! I have a few small comments (see inline). Also a few additional thoughts:
* can you also add Spain-specific tests? See [Japanese tests](https://github.com/eprbell/rp2/blob/main/tests/test_ods_output_diff_jp.py) as an example;
* to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;
* after this PR is merged, could you add a [ES plugin to DaLI](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#country-plugin-development) as well (again, you can see the DaLI Japanese plugin as a guide)? It's trivial to implement, but it has a big impact because it makes DaLI available to Spanish users.

Thanks! And if you have any questions, just let me know.

File: src/rp2/plugin/report/es/open_positions.py (hunk 6 above)
Comment (Larision) [reply src/rp2/plugin/report/es/open_positions.py:1]: Checking about it, all languages showing $ in currency using common open_positions.
Thats why i added ES specific template_open_positions_es.ods and es.open_positions.py.

**Comment by Larision** [comment]
> * to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;

can you help me a bit more? what code i need update?

**Comment by eprbell** [comment]
> > * to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;
> 
> can you help me a bit more? what code i need update?



> > * to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;
> 
> can you help me a bit more? what code i need update?

Basically you need to bring the latest changes from the main branch into your branch: take a look at https://stackoverflow.com/questions/59916041/how-to-update-a-branch-with-master-on-github

File: src/rp2/plugin/report/es/open_positions.py (hunk 6 above)
Comment (eprbell) [reply src/rp2/plugin/report/es/open_positions.py:1]: Oh, I see: let me take a look at this later today.

**Comment by Larision** [comment]
> > > * to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;
> > 
> > 
> > can you help me a bit more? what code i need update?
> 
> Basically you need to bring the latest changes from the main branch into your branch: take a look at https://stackoverflow.com/questions/59916041/how-to-update-a-branch-with-master-on-github

my repo is already synced with yours. checked twice. im lost with this :(

File: src/rp2/plugin/report/es/open_positions.py (hunk 6 above)
Comment (eprbell) [reply src/rp2/plugin/report/es/open_positions.py:1]: OK, I fixed the issue: if you get the sync your branch to the latest version you'll be pick up the fix (see my other comment on how to do that).

**Comment by eprbell** [comment]
> > > > * to fix the documentation check error in CI: can you just update your code? I fixed it and you just need to pick up the latest version of the code;
> > > 
> > > 
> > > can you help me a bit more? what code i need update?
> > 
> > 
> > Basically you need to bring the latest changes from the main branch into your branch: take a look at https://stackoverflow.com/questions/59916041/how-to-update-a-branch-with-master-on-github
> 
> my repo is already synced with yours. checked twice. im lost with this :(

Sorry, my bad! I had disabled the Twitter link check in DaLI, but not in RP2: I just fixed it now. Can you sync one more time? It should pick up the documentation link fix and also the open_positions fix. Let me know if you have any problems.

**Comment by Larision** [comment]
> * after this PR is merged, could you add a [ES plugin to DaLI](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#country-plugin-development) as well (again, you can see the DaLI Japanese plugin as a guide)? It's trivial to implement, but it has a big impact because it makes DaLI available to Spanish users.

Already done, but i have same problem with template and € symbol i had here in rp2 with open_positions

File: src/rp2/plugin/country/es.py (hunk 5 above)
Comment (eprbell) [reply src/rp2/plugin/country/es.py:42]: This is still unmodified (I think this is why you're still seeing the wrong currency). You need to:
* remove `es.` and 
* delete src/rp2/plugin/report/es/open_positions.py

File: tests/test_ods_output_diff_es.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 Neal Chambers
```
Comment (eprbell) [inline tests/test_ods_output_diff_es.py:1]: Use your name in the copyright line.

**Review by eprbell — CHANGES_REQUESTED** [review]
A couple more changes are needed.

**Comment by eprbell** [comment]
> > * after this PR is merged, could you add a [ES plugin to DaLI](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#country-plugin-development) as well (again, you can see the DaLI Japanese plugin as a guide)? It's trivial to implement, but it has a big impact because it makes DaLI available to Spanish users.
> 
> Already done, but i have same problem with template and € symbol i had here in rp2 with open_positions

I'm not seeing a PR in DaLI about this: can you upload it?

**Comment by Larision** [comment]
> > > * after this PR is merged, could you add a [ES plugin to DaLI](https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#country-plugin-development) as well (again, you can see the DaLI Japanese plugin as a guide)? It's trivial to implement, but it has a big impact because it makes DaLI available to Spanish users.
> > 
> > 
> > Already done, but i have same problem with template and € symbol i had here in rp2 with open_positions
> 
> I'm not seeing a PR in DaLI about this: can you upload it?

I was waiting this PR close to upload it

**Review by eprbell — CHANGES_REQUESTED** [review]
Looks pretty good! One thing I noticed is that transaction types are not translated in rp2_full_report output. I fixed that in the latest version (see [b8294df](https://github.com/eprbell/rp2/commit/b8294df26bc4f00f84e44775047f85a1c5900054)).

To get the fix you need to sync the code again, then [regenerate localization strings](https://github.com/eprbell/rp2/blob/main/README.dev.md#localization) and the translations for transaction types.

**Comment by Larision** [comment]
> Looks pretty good! One thing I noticed is that transaction types are not translated in rp2_full_report output. I fixed that in the latest version (see [b8294df](https://github.com/eprbell/rp2/commit/b8294df26bc4f00f84e44775047f85a1c5900054)).
> 
> To get the fix you need to sync the code again, then [regenerate localization strings](https://github.com/eprbell/rp2/blob/main/README.dev.md#localization) and the translations for transaction types.

Synced, regenerated and tested, not working in transaction type neither in taxable event direction/type column.

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -631,6 +630,46 @@ msgstr "Resumen de ganancia / pérdida anual"
...
+msgid "gift"
+msgstr "regalo"
+
+msgid "hardfork"
+msgstr "bifurcación dura"
+
+msgid "income"
+msgstr "impuesto"
+
+#, fuzzy
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:654]: Remove the fuzzy comment.

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -631,6 +630,46 @@ msgstr "Resumen de ganancia / pérdida anual"
 msgid "Yellow"
 msgstr "Amarillo"
 
+msgid "airdrop"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:634]: Missing translation: is this intended?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -631,6 +630,46 @@ msgstr "Resumen de ganancia / pérdida anual"
...
+
+msgid "income"
+msgstr "impuesto"
+
+#, fuzzy
+msgid "interest"
+msgstr "interes"
+
+msgid "mining"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:659]: Missing translation: is this intended?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -631,6 +630,46 @@ msgstr "Resumen de ganancia / pérdida anual"
...
+
+msgid "move"
+msgstr "movimiento"
+
+msgid "sell"
+msgstr "venta"
+
+msgid "staking"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:668]: Missing translation: is this intended?

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 12]
```diff
@@ -630,6 +630,48 @@ msgstr "__test_Yearly Gain / Loss Summary"
 msgid "Yellow"
 msgstr "__test_Yellow"
 
+msgid "airdrop"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:634]: This should be `__test_airdrop`. Same for the ones below: they all start with `__test_` followed by the English word.

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -630,6 +630,48 @@ msgstr "__test_Yearly Gain / Loss Summary"
...
+
+msgid "buy"
+msgstr ""
+
+msgid "donate"
+msgstr ""
+
+#, fuzzy
+msgid "fee"
+msgstr "__test_{} Fee"
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:644]: Should be `__test_fee`

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 14]
```diff
@@ -630,6 +630,48 @@ msgstr "__test_Yearly Gain / Loss Summary"
...
+msgid "airdrop"
+msgstr ""
+
+msgid "buy"
+msgstr ""
+
+msgid "donate"
+msgstr ""
+
+#, fuzzy
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:642]: Remove fuzzy comment.

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -630,6 +630,48 @@ msgstr "__test_Yearly Gain / Loss Summary"
...
+msgid "gift"
+msgstr ""
+
+msgid "hardfork"
+msgstr ""
+
+msgid "income"
+msgstr ""
+
+#, fuzzy
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:655]: Remove fuzzy comment.

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -630,6 +630,48 @@ msgstr "__test_Yearly Gain / Loss Summary"
...
+msgid "interest"
+msgstr "__test_Notes"
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:657]: Should be `__test_interest`

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 17]
```diff
@@ -711,6 +753,13 @@ msgstr "__test_{} value of the transaction with fees"
 msgid "{} value of the transaction without fees"
 msgstr "__test_{} value of the transaction without fees"
 
+#, fuzzy
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:756]: Remove fuzzy comment.

File: src/rp2/locales/kl/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -711,6 +753,13 @@ msgstr "__test_{} value of the transaction with fees"
...
+msgid "{}_Summary"
+msgstr "__test_Summary"
```
Comment (eprbell) [inline src/rp2/locales/kl/LC_MESSAGES/messages.po:758]: Should be `__test_{}_Summary`

**Review by eprbell — CHANGES_REQUESTED** [review]
I think we're almost there! Just a few small changes and then we can merge.

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 9 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:634]: intended, no good translation in spanish.

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 11 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:668]: intended, no good translation.

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 8 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:654]: removed

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 10 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:659]: intended but translation it now.

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 19]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "Precio medio pagado por 1 crypto"
+
+msgid "Average price at which the crypto was acquired"
+msgstr "Precio medio al que la criptomoneda fue adquirido"
+
+msgid "Average price at which the crypto was acquired, across all transactions"
+msgstr "Precio medio al que se adquirió la criptomoneda, en todas las transacciones"
+
+msgid "Balance"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:103]: Missing translation?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 20]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "Saldo de todas las transacciones de BUY y EARN para un intercambio y titular dados"
+
+msgid "Balance of all the SEND and SOLD transactions for which the given exchange and holder are receiver"
+msgstr "Saldo de todas las transacciones SEND y SOLD para las cuales el intercambio y el titular dados son el receptor"
+
+msgid "Balance of all the SEND and SOLD transactions for which the given exchange and holder are sender"
+msgstr "Saldo de todas las transacciones SEND y SOLD para las cuales el intercambio y el titular dados son el remitente"
+
+msgid "Capital"
+msgstr ""
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:115]: Even though the Spanish word is the same as the English one here, I suggest filling the string, in the (unlikely) case the English string is changed. Empty translation strings should be used only for English words that don't have a Spanish translation and are commonly used without translation.

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 21]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "¿La transacción contiene un evento imponible? Si es así, la cantidad imponible se resalta en amarillo"
+
+msgid "EARN (crypto earned through interest, etc.), GIFT (crypto given), SOLD (crypto sold) OR INTRA (fees for transferring crypto across accounts)"
+msgstr "EARN (cripto ganada a través de intereses, etc.), GIFT (cripto dado), SOLD (cripto vendido) O INTRA (comisiónes por transferir cripto entre cuentas)"
+
+msgid "ENTER PRICES ON INPUT TAB"
+msgstr "INGRESE PRECIOS EN LA PESTAÑA DE ENTRADA"
+
+msgid "English description of this in-transaction fraction"
+msgstr "Descripción en inglés de esta fracción en transacción"
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:223]: This is the description of the Notes column: should it be "en español"?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 22]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "Importe crypto"
+
+msgid "Crypto Amt Running Sum"
+msgstr "Importe crypto suma acumulada"
+
+msgid "Crypto Balance"
+msgstr "Balance crypto"
+
+msgid "Crypto Balance * Input Price"
+msgstr "Balance crypto * Precio de entrada"
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:163]: Same: saldo?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 23]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "Sentido (ENTRADA/SALIDA/INTRA) y tipo (COMPRA, VENDIDA, etc) del hecho imponible"
+
+msgid "Direction/Type"
+msgstr "Dirección/Tipo"
+
+msgid "Does the transaction contain a taxable event? If so the taxable amount is highlighted in yellow"
+msgstr "¿La transacción contiene un evento imponible? Si es así, la cantidad imponible se resalta en amarillo"
+
+msgid "EARN (crypto earned through interest, etc.), GIFT (crypto given), SOLD (crypto sold) OR INTRA (fees for transferring crypto across accounts)"
+msgstr "EARN (cripto ganada a través de intereses, etc.), GIFT (cripto dado), SOLD (cripto vendido) O INTRA (comisiónes por transferir cripto entre cuentas)"
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:217]: Translate the English words here?

File: src/rp2/locales/es/LC_MESSAGES/messages.po [hunk 24]
```diff
@@ -0,0 +1,757 @@
...
+msgstr "Porcentaje del costo base total"
+
+msgid "Transaction"
+msgstr "Transacción"
+
+msgid "Transaction Type"
+msgstr "Tipo de transacción"
+
+msgid "Transactions that added new crypto to accounts (e.g. buy, etc.). Only EARN-typed transactions are taxable events (interest received on crypto)"
+msgstr "Transacciones que agregaron nueva criptomoneda a las cuentas (compra, etc.). Solo las transacciones de tipo EARN son eventos imponibles (intereses recibidos en criptomoneda)."
```
Comment (eprbell) [inline src/rp2/locales/es/LC_MESSAGES/messages.po:562]: Translate EARN?

**Review by eprbell — CHANGES_REQUESTED** [review]
A few more nits.

**Comment by Larision** [comment]
```
msgid "USD"
msgstr "USD"

msgid "USD Total"
msgstr "Total en USD"
```
Should i change USD for EUR?

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 22 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:163]: right, updated

File: src/rp2/locales/es/LC_MESSAGES/messages.po (hunk 24 above)
Comment (Larision) [reply src/rp2/locales/es/LC_MESSAGES/messages.po:562]: i left EARN without translation, hard to translate with right meaning.

**Comment by eprbell** [comment]
> ```
> msgid "USD"
> msgstr "USD"
> 
> msgid "USD Total"
> msgstr "Total en USD"
> ```
> 
> Should i change USD for EUR?

Yes, good catch.

**Review by eprbell — APPROVED** [review]
LGTM: this is a really nice addition, thanks for the PR, @Larision, and welcome, Spain, to RP2!

**Pull request merged**